src/components/organisms/teacher/OakUnitListOptionalityItem/OakUnitListOptionalityItem.tsx

```diff
@@ -0,0 +1,244 @@
+        $font={"heading-7"}
+        tag="h3"
+      >
+        {nullTitle}
+      </OakHeading>
+    </OakFlex>
+  );
+};
+
+export type OakUnitListOptionalityItemProps<C extends ElementType> = {
+  as?: C;
+  disabled?: boolean;
+  unavailable?: boolean;
+  index: number;
+  nullTitle: string | undefined;
+  yearTitle?: string | null;
+
+  optionalityUnits: {
+    title: string;
+    href: string;
+
+    lessonCount: number;
+    firstItemRef?:
+      | MutableRefObject<HTMLAnchorElement | null>
+      | null
+      | undefined;
+  }[];
+} & ComponentPropsWithoutRef<C>;
+
+/**
+ *
+ * OakUnitsListOptionalityItem component used as links for unit cards with optionality
+ */
+export const OakUnitListOptionalityItem = <C extends ElementType>(
+  props: OakUnitListOptionalityItemProps<C>,
+) => {
+  const {
+    as,
+    lessonSectionName,
+    lessonCount,
+    progress,
+    disabled,
+    href,
+    unavailable,
+    onClick,
+    index,
+    firstItemRef,
+    optionalityUnits,
+    nullTitle,
+    ...rest
+  } = props;
+
+  const disabledOrUnavailable = disabled || unavailable;
+
+  return (
+    <OakFlex
+      $flexDirection={["column", "row", "row"]}
+      $width={"100%"}
+      role="listitem"
+      {...rest}
+    >
+      <OakFlex $display={["flex", "none"]} $background={"white"}>
+        <StyledIndex
+          index={index}
+          disabledOrUnavailable={disabledOrUnavailable}
+        />
+
+        <OakFlex $alignItems={"center"} $ma={"space-between-xs"}>
+          <StyledUnitHeading
+            nullTitle={nullTitle}
+            disabledOrUnavailable={disabledOrUnavailable}
+          />
+        </OakFlex>
+      </OakFlex>
+      <StyledIndex
+        index={index}
+        disabledOrUnavailable={disabledOrUnavailable}
+        $display={["none", "flex", "flex"]}
+      />
+
+      <OakBox
+        $background={"white"}
+        $pa={["inner-padding-m", "inner-padding-l"]}
+        $width={"100%"}
+      >
+        <OakFlex
+          $alignItems={"center"}
+          $mb={"space-between-s"}
+          $display={["none", "none", "flex"]}
+        >
+          <OakFlex
+            $flexGrow={1}
+            $alignItems={"center"}
+            $justifyContent={"space-between"}
+          >
+            <StyledUnitHeading
+              nullTitle={nullTitle}
+              disabledOrUnavailable={disabledOrUnavailable}
+            />
+            <OakFlex $alignItems={"center"}>
+              <StyledYearAndOptionCount
+                yearTitle={props.yearTitle}
+                optionalityUnitsLength={optionalityUnits.length}
+                disabledOrUnavailable={disabledOrUnavailable}
+              />
+            </OakFlex>
+          </OakFlex>
+        </OakFlex>
+        <OakBox $mb={"space-between-s"} $display={["none", "flex", "none"]}>
+          <StyledUnitHeading
+            nullTitle={nullTitle}
+            disabledOrUnavailable={disabledOrUnavailable}
+          />
+        </OakBox>
```

Comment:
Do we still want to leave the space for the margin when `nullTitle` is `undefined` and the element doesn't render?